pkg/types/common/common.go

```diff
@@ -0,0 +1,66 @@
+package common
```

Comment:
This is only for validation right? Maybe move to the pkg/types/validation package, rather than common

Reply:
That would create a cyclic dependence as this code is being used from `baremetal/validation` as well.